### DISCUSSION TITLE
[Question] Implementation of a pushbutton

### DISCUSSION THREAD

**Discussion by vpoguru** [GrandOrgue/grandorgue#831] [Help] [answered]
I want to implement a pushbutton (momentary switch) for a bell. The _Pushbutton object_ mentioned in the help file obviously does not exist.

I created a reversible piston that activates a stop that contains the bell sound:

    [ReversiblePiston001]
    Name=Bell
    ManualNumber=003 ; Effects
    ObjectType=Stop
    ObjectNumber=002 ; Bell

In the [Panel000] I created a link to the reversible piston:

    [Panel000Element027]
    Type=ReversiblePiston
    ReversiblePiston=001 ; Bell
    ...

This implementation works exactly like a stop that is activated via a latching switch. I have to manually reset the pushbutton to the OFF state.

What is the technique to implement a pushbutton (or momentary switch)?

**Comment by rousseldenis** [comment]
Hi @vpoguru, maybe @larspalo can answer to this, but after having read the documentation, I don't think this is possible.

**Comment by larspalo** [comment, marked as answer]
Well, as always there can be many ways to do things, and there's absolutely nothing wrong doing it as you've done... 

In the GUI there's no way to mimic a momentary switch behaviour (for instance: one mouse click would show a on-image for some time, or as long as the mouse is held, and then an off-image automatically or when mouse is release). With MIDI signals it's easily done from a playing console without resorting to tricks. This was a deliberate design choice as GO was always intended to be played from a MIDI console and not only from a computer screen.

However, it's possible to have both on and off state make a sound, and if the image would be the same or whatever it can look differently depending on what you want. I attach a very small example of how I've done with others sound effects when they don't contain any loop and cue markers and I still want to have sound both on and off. (Which in your case could mean that every time the element is clicked you'll have a sound)

Personally, when I make them myself, I treat such effect stop samples like any other sample and set a short loop in a silent part just before the release (either included in the same sample file or loaded separately).

[EffectStop.zip](https://github.com/GrandOrgue/grandorgue/files/7455746/EffectStop.zip)

**Comment by vpoguru** [comment]
Thank you for the detailed answer! Maybe it is an option to implement momentary switches in the future...
